pkg/apis/pipeline/v1beta1/pipelineparams_types.go

```diff
@@ -22,21 +22,16 @@ import (
 
 // PipelineParamsSpec is the spec for a Pipeline resource
 type PipelineParamsSpec struct {
-	ServiceAccount string          `json:"serviceAccount"`
-	Sources        []Source        `json:"sources"`
-	ArtifactStores []ArtifactStore `json:"artifactStores"`
-	Results        Results         `json:"results"`
+	ServiceAccount string  `json:"serviceAccount"`
+	Results        Results `json:"results"`
 }
 
-// Source is an endpoint from which to get data which is required
-// by a Build/Task for context (e.g. a repo from which to build an image).
-type Source struct {
-	Name           string `json:"name"`
-	Type           string `json:"type"`
-	URL            string `json:"url"`
-	Branch         string `json:"branch"`
-	Commit         string `json:"commit,omitempty"`
-	ServiceAccount string `json:"serviceAccount,omitempty"`
+type Source interface {
+	getName() string
+	getType() string
+	getParams() []Param
+	getVersion() string
+	getPassingConstraints() []TaskRef
```

Comment:
Can you add more information on this method?
getPassingConstraints()

Reply:
good catch, that should not be here.  We first thought about putting the constraint about a Source being passed from one task to another here, but moved it to SourceBinding. Missed removing it from here. Where do you think is a better place for that?

Reply:
I feel, `getPassingConstrainsts` should be defined at the interface level. 
The reason being, for each source type, they are going to be constant. You can multiple sources of same source type defined in your cluster. You do not want to repeat the binding information for each  sourcebinding.
Does that make sense?